pkg/http-handler/http-handler.go

```diff
@@ -75,6 +76,34 @@ func getNamespaceAndCluster(req *http.Request) (string, string) {
 	return requestNamespace, requestCluster
 }
 
+// GetHeaderNamespaces returns a list of namespaces from the header request
+// The name and the value of the header field is specified by 2 environment variables:
+// - KUBEAPPS_NAMESPACE_HEADER is a name of the expected header field,
+//     e.g. KUBEAPPS_NAMESPACE_HEADER=X-Consumer-Groups
+// - KUBEAPPS_NAMESPACE_PATTERN is a regular expression and it matches only single regex group,
+//     e.g. KUBEAPPS_NAMESPACE_PATTERN=namespace:([\w-]+)
+func getHeaderNamespaces(req *http.Request) []string {
+	headerName := os.Getenv("KUBEAPPS_NAMESPACE_HEADER")
+	if headerName == "" {
+		return []string{}
+	}
+	headerPattern := os.Getenv("KUBEAPPS_NAMESPACE_PATTERN")
+	if headerPattern == "" {
+		return []string{}
+	}
+	r := regexp.MustCompile(headerPattern)
```

Comment:
`MustCompile` panics if the pattern is not valid, better to use `Compile` and gracefully handle the error (mostly because we cannot be sure the patter is valid).